packages/app-store/ee/routing-forms/trpc-router.ts

```diff
@@ -41,24 +99,21 @@ const app_RoutingForms = createRouter()
               code: "NOT_FOUND",
             });
           }
-          const fieldsParsed = zodFields.safeParse(form.fields);
-          if (!fieldsParsed.success) {
-            // This should not be possible normally as before saving the form it is verified by zod
-            throw new TRPCError({
-              code: "INTERNAL_SERVER_ERROR",
-            });
-          }
-
-          const fields = fieldsParsed.data;
 
-          if (!fields) {
+          const serializableForm = getSerializableForm(form);
+          if (!serializableForm.fields) {
             // There is no point in submitting a form that doesn't have fields defined
             throw new TRPCError({
               code: "BAD_REQUEST",
             });
           }
 
-          const missingFields = fields
+          const serializableFormWithFields = {
```

Comment:
This creates a type that has `fields` set unlike `serializableForm` which has fields optional.